deployment/src/main/java/com/datastax/oss/quarkus/deployment/internal/CassandraClientProcessor.java

```diff
@@ -64,76 +64,67 @@ FeatureBuildItem feature() {
 
   @BuildStep
   List<ReflectiveClassBuildItem> registerGraphForReflection() {
-    return Arrays.asList(
-        // Required for the driver DependencyCheck mechanism
-        new ReflectiveClassBuildItem(
-            true, true, "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal"),
-        // Should be initialized at build time:
-        new ReflectiveClassBuildItem(
-            true, true, "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0"),
-        new ReflectiveClassBuildItem(
-            true, true, "org.apache.tinkerpop.shaded.jackson.databind.deser.std.StdDeserializer"),
-        // Required by Tinkerpop:
-        // TODO check if this is really all that is instantiated by reflection
-        new ReflectiveClassBuildItem(true, true, "org.apache.tinkerpop.gremlin.structure.Graph"),
-        new ReflectiveClassBuildItem(
-            true, true, "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph"),
-        new ReflectiveClassBuildItem(
-            true, true, "org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph"),
-        new ReflectiveClassBuildItem(
-            true,
-            true,
-            "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource"));
+
+    /* TODO: We're creating an RCBI instance per class here to match up with the pre-Quarkus3.0 usage.
+     * Check to see if we can just simplify this to something like RCBI.builder(names).foo().build() */
+    return Stream.of(
+            "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal",
+            "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0",
+            "org.apache.tinkerpop.shaded.jackson.databind.deser.std.StdDeserializer",
+            // Required by Tinkerpop:
+            // TODO check if this is really all that is instantiated by reflection
+            "org.apache.tinkerpop.gremlin.structure.Graph",
+            "org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph",
+            "org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph",
+            "org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource")
+        .map(
+            (clzName) ->
+                ReflectiveClassBuildItem.builder(clzName).methods(true).fields(true).build())
+        .collect(Collectors.toList());
```

Comment:
This implementation deliberately mirrors the previous approach of creating a distinct RCBI instance for every class name we wish to account for.  The [relevant RCBI impl](https://github.com/quarkusio/quarkus/blob/3.0.0.Final/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java) seems to suggest that we _might_ be able to accomplish the same thing with a single RCBI instance fed with multiple class names.  In the interest of expediency I'm not pursuing that path at the moment... that's what the TODO above is meant to remind me about.